tests/Feature/LMSJoinClassTest.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace Uasoft\Badaso\Module\LMSModule\Tests\Feature;
+
+use Uasoft\Badaso\Module\LMSModule\Models\Course;
+use Uasoft\Badaso\Module\LMSModule\Models\CourseUser;
+use Uasoft\Badaso\Module\LMSModule\Models\User;
+use Tests\TestCase;
+use Uasoft\Badaso\Module\LMSModule\Tests\Helpers\AuthHelper;
+
+class LMSJoinClassTest extends TestCase
+{
+
+  public function testJoinClassWithoutLoginExpectResponseStatus401()
+  {
+    $url = route('badaso.course.join');
+    $response = $this->json('POST', $url);
+    $response->assertStatus(401);
+  }
+
+  public function testJoinClassAsAuthorizedUserWithUnknownClassCodeExpectResponseStatus500()
+  {
+      $user = User::factory()->create();
+      $user->rawPassword = 'password';
+
+      $url = route('badaso.course.join');
+
+      $response = AuthHelper::asUser($this, $user)->json('POST', $url, [
+          'code' => 'xxx'
+      ]);
+      $response->assertStatus(500);
+  }
+
+  public function testJoinClassAsAuthorizedUserWithNoClassCodeAsInputExpectResponseStatus400()
+  {
+      $user = User::factory()->create();
+      $user->rawPassword = 'password';
+
+      $url = route('badaso.course.join');
+
+      $response = AuthHelper::asUser($this, $user)->json('POST', $url, []);
+      $response->assertStatus(400);
+  }
+
+  public function testJoinClassAsAuthorizedUserWithValidClassCodeExpectResponseStatus200()
+  {
+      $user_teacher = User::factory()->create();
+      $user_teacher->rawPassword = 'password';
+
+      $user_student = User::factory()->create();
+      $user_student->rawPassword = 'password';
```

Comment:
Please do use camelCase for variable naming. Just to make it consistent.
ref: https://xqsit.github.io/laravel-coding-guidelines/docs/naming-conventions/